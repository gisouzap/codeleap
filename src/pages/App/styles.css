:root {
  --color-white: #fff;
  --color-black: #000;
  --color-primary: #7695ec;
  --color-danger: #ff5151;
  --color-success: #47b960;
  --color-grey-100: #dddddd;
  --color-grey-200: #cccccc;
  --color-grey-300: #777777;
  --color-grey-400: #999999;
}

* {
  font-family: 'Roboto', sans-serif;
  box-sizing: border-box;
  margin: 0px;
  color: #000;
}

::-webkit-scrollbar {
  width: 4px;
}

::-webkit-scrollbar-track {
  background-color: #f5f5f5;
}

::-webkit-scrollbar-thumb {
  background-color: var(--color-primary);
  border-radius: 5px;
}

::-moz-scrollbar {
  width: 4px;
}

::-moz-scrollbar-track {
  background-color: #f5f5f5;
}

::-moz-scrollbar-thumb {
  background-color: var(--color-primary);
  border-radius: 5px;
}

::-ms-scrollbar {
  width: 4px;
}

::-ms-scrollbar-track {
  background-color: #f5f5f5;
}

::-ms-scrollbar-thumb {
  background-color: var(--color-primary);
  border-radius: 5px;
}
