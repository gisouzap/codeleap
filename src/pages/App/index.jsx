import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import Root from '../Root';

import './styles.css';
const history = createBrowserHistory();

function App() {
  return (
    <BrowserRouter history={history}>
      <Routes>
        <Route exact path="/" Component={Root} />
      </Routes>
    </BrowserRouter>
  );
}

export default App;
