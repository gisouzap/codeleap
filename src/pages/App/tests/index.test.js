import React from 'react';
import { render } from '@testing-library/react';
import App from '../index';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';

describe('App', () => {
  const mockStore = configureStore([]);

  const store = mockStore({
    user: { login: { isLoggedIn: true } },
  });

  it('should not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');

    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('should render and match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
