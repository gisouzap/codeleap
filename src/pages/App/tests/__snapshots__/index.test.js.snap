// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render and match the snapshot 1`] = `
<div>
  <div
    class="signup"
  >
    <div
      class="signup-wrapper"
    >
      <div
        class="card"
      >
        <p
          class="title"
        >
          Welcome to CodeLeap network!
        </p>
        <div
          class="card-input"
        >
          <p
            class="label"
          >
            Please enter your username
          </p>
          <input
            class="input"
            placeholder="John doe"
            type="text"
            value=""
          />
        </div>
        <div
          class="card-button"
        >
          <button
            class="button primary"
            disabled=""
            type="button"
          >
            Enter
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
